src/Orleans/Transactions/TransactionContext.cs

```diff
@@ -90,13 +91,40 @@ public TransactionInfo(TransactionInfo other)
 
         public bool IsAborted { get; set; }
 
-        public int PendingCalls { get; set; }
 
         public Dictionary<ITransactionalResource, TransactionalResourceVersion> ReadSet { get; }
         public Dictionary<ITransactionalResource, int> WriteSet { get; }
         public HashSet<long> DependentTransactions { get; }
 
-        public void Union(TransactionInfo other)
+        [NonSerialized]
+        public int PendingCalls;
+
+        [NonSerialized]
+        ConcurrentQueue<TransactionInfo> joined = new ConcurrentQueue<TransactionInfo>();
+        // protect from null ref in case join is accessed on serialized object.
+        private ConcurrentQueue<TransactionInfo> Joined => this.joined ?? (this.joined = new ConcurrentQueue<TransactionInfo>());
```

Comment:
can this ever be null? you are assigning it in the field initializer... does it not get initialized too when deserializing?

Reply:
Unfortunately non-serialized fields are not initialized when deserializing, so yes, this can be null :(

Reply:
In this case, that's because there's no parameterless constructor, so the object must be created using `GetUninitializedObject` during deserialization.

Reply:
@ReubenBond Empty constructor is not a valid use case.  Added one for serialization purposes.